zkevm-circuits/src/evm_circuit.rs

```diff
@@ -1 +1,699 @@
+            Vec::<(Expression<F>, Vec<Lookup<F>>)>::new();
+
+        let op_execution_gadget = OpExecutionGadget::configure(
+            meta,
+            r,
+            qs_op_execution,
+            qs_byte_lookups,
+            op_execution_state_curr,
+            op_execution_state_next,
+            op_execution_free_cells,
+            &mut independent_lookups,
+        );
+
+        // TODO: call_initialization_gadget
+
+        Self::configure_lookups(
+            meta,
+            qs_byte_lookup,
+            advices,
+            fixed_table,
+            independent_lookups,
+        );
+
+        EvmCircuit {
+            q_step,
+            qs_byte_lookup,
+            fixed_table,
+            op_execution_gadget,
+        }
+    }
+
+    // TODO: refactor return type
+    #[allow(clippy::type_complexity)]
+    fn configure_allocations(
+        meta: &mut ConstraintSystem<F>,
+        q_step: Selector,
+        qs_byte_lookup: Column<Advice>,
+        advices: [Column<Advice>; CIRCUIT_WIDTH],
+    ) -> (
+        Expression<F>,
+        Vec<Cell<F>>,
+        OpExecutionState<F>,
+        OpExecutionState<F>,
+        Vec<Cell<F>>,
+    ) {
+        let mut qs_byte_lookups = Vec::with_capacity(CIRCUIT_HEIGHT);
+        meta.create_gate("Query byte lookup as a synthetic selector", |meta| {
+            for h in 0..CIRCUIT_HEIGHT as i32 {
+                qs_byte_lookups.push(Cell {
+                    expression: meta.query_advice(qs_byte_lookup, Rotation(h)),
+                    column: qs_byte_lookup,
+                    rotation: h as usize,
+                });
+            }
+
+            vec![Expression::Constant(F::zero())]
+        });
+
+        // TODO: Enable this when switch to kzg branch and remove the failure
+        //       VerifyFailure::ConstraintPoison
+        // meta.create_gate("Query byte lookup as a synthetic selector", |meta| {
+        //     let exp = meta.query_advice(qs_byte_lookup, Rotation::cur());
+
+        //     vec![exp.clone() * (Expression::Constant(F::one()) - exp)]
+        // });
+
+        let mut cells_curr = Vec::with_capacity(CIRCUIT_WIDTH * CIRCUIT_HEIGHT);
+        meta.create_gate("Query cells for current step", |meta| {
+            for h in 0..CIRCUIT_HEIGHT as i32 {
+                for &column in advices.iter() {
+                    cells_curr.push(Cell {
+                        expression: meta.query_advice(column, Rotation(h)),
+                        column,
+                        rotation: h as usize,
+                    });
+                }
+            }
+
+            vec![Expression::Constant(F::zero())]
+        });
+
+        let num_cells_next_asseccible = NUM_CELL_OP_EXECTION_STATE;
+
+        let cells_next =
+            &mut cells_curr[..num_cells_next_asseccible].to_vec()[..];
+        meta.create_gate("Query cells for next step", |meta| {
+            for cell in cells_next.iter_mut() {
+                cell.expression = meta.query_advice(
```

Comment:
`cell.rotation` is not updated in the `cells_next`

Reply:
I think it might be easier to just create new `Cell` for `cells_next`.

Reply:
Thanks for catching this! Fixed in [`4c9dc9d`](https://github.com/appliedzkp/zkevm-circuits/pull/26/commits/4c9dc9d33ac28d9f7eaac2af389d52aa02ad146e)